code/test/sixsq/nuvla/server/resources/infrastructure_service_generic_lifecycle_test.clj

```diff
@@ -131,4 +131,4 @@
         (-> session
             (request abs-uri :request-method :delete)
             (ltu/body->edn)
-            (ltu/is-status 200))))))
+            (ltu/is-status 202))))))
```

Comment:
This is the wrong return code. A 202 response should only be provided when a job is created.  There is no need to create a job in this case.

Reply:
but there is a job which is created on delete right?

Reply:
that's true, and it doesn't...so you're saying in https://github.com/nuvla/api-server/blob/92c14530970960f8aa4b664108ff07a9deb02d4a/code/src/sixsq/nuvla/server/resources/infrastructure_service.clj#L202, it should retunr 200?

Reply:
The `202 accepted` response is only to be used when a job is created and the action is being handled asynchronously.  If the action is handled synchronously, then `200` should be returned on success or another appropriate error code on failure.

Reply:
but, because of the `if` statement, we know that this will be handled asynchronously , so that line will always be 202

Reply:
changed the tests to delete only when the state is valid for deletion